pkg/runtime/interfaces.go

```diff
@@ -20,29 +20,23 @@ import (
 	"io"
 )
 
-// ObjectScheme represents common conversions between formal external API versions
-// and the internal Go structs. ObjectScheme is typically used with ObjectCodec to
-// transform internal Go structs into serialized versions. There may be many valid
-// ObjectCodecs for each ObjectScheme.
-type ObjectScheme interface {
-	ObjectConvertor
-	ObjectTyper
-	ObjectCreater
-	ObjectCopier
-}
-
-// ObjectCodec represents the common mechanisms for converting to and from a particular
-// binary representation of an object.
-type ObjectCodec interface {
-	ObjectEncoder
+// Codec defines methods for serializing and deserializing API objects.
+type Codec interface {
 	Decoder
+	Encoder
+
+	// Methods for processing url parameters.
+	// EncodeParameters(obj Object) (url.Values, obj Object) error
+	// DecodeParametersInto(parameters url.Values, obj Object) error
 }
 
 // Decoder defines methods for deserializing API objects into a given type
 type Decoder interface {
 	Decode(data []byte) (Object, error)
+	// TODO: Remove this method?
 	DecodeToVersion(data []byte, version string) (Object, error)
 	DecodeInto(data []byte, obj Object) error
+	// TODO: Remove this method?
 	DecodeIntoWithSpecifiedVersionKind(data []byte, obj Object, kind, version string) error
```

Comment:
But yes. Codec parameters should not be referencing group/version/kind. I didn't stop `DecodeIntoWithSpecifiedVersionKind` because we needed to make progress.